docs/JAX_time_series_classification.md

```diff
@@ -0,0 +1,389 @@
+)
+
+test_sampler = grain.IndexSampler(
+    len(test_source),
+    shuffle=False,
+    seed=seed,
+    shard_options=grain.NoSharding(),  # No sharding since this is a single-device setup
+    num_epochs=1,                      # Iterate over the dataset for one epoch
+)
+
+
+train_loader = grain.DataLoader(
+    data_source=train_source,
+    sampler=train_sampler,  # Sampler to determine how to access the data
+    worker_count=4,         # Number of child processes launched to parallelize the transformations among
+    worker_buffer_size=2,   # Count of output batches to produce in advance per worker
+    operations=[
+        grain.Batch(train_batch_size, drop_remainder=True),
+    ]
+)
+
+test_loader = grain.DataLoader(
+    data_source=test_source,
+    sampler=test_sampler,  # Sampler to determine how to access the data
+    worker_count=4,        # Number of child processes launched to parallelize the transformations among
+    worker_buffer_size=2,  # Count of output batches to produce in advance per worker
+    operations=[
+        grain.Batch(test_batch_size),
+    ]
+)
+```
+
+## Define the Model
+
+Let's now construct the Convolutional Neural Network with Flax by subclassing `nnx.Module`.
+You can learn more about the [Flax NNX module system in the Flax documentation](https://flax.readthedocs.io/en/latest/nnx_basics.html#the-flax-nnx-module-system).
+
+Let's have three convolution and dense layers, and use ReLU activation function for middle
+layers and softmax in the final layer for binary classification output.
+
+```{code-cell} ipython3
+class MyModel(nnx.Module):
+    def __init__(self, rngs: nnx.Rngs):
+        self.conv_1 = nnx.Conv(
+            in_features=1, out_features=64, kernel_size=3, padding="SAME", rngs=rngs
+        )
+        self.layer_norm_1 = nnx.LayerNorm(num_features=64, epsilon=0.001, rngs=rngs)
+
+        self.conv_2 = nnx.Conv(
+            in_features=64, out_features=64, kernel_size=3, padding="SAME", rngs=rngs
+        )
+        self.layer_norm_2 = nnx.LayerNorm(num_features=64, epsilon=0.001, rngs=rngs)
+
+        self.conv_3 = nnx.Conv(
+            in_features=64, out_features=64, kernel_size=3, padding="SAME", rngs=rngs
+        )
+        self.layer_norm_3 = nnx.LayerNorm(num_features=64, epsilon=0.001, rngs=rngs)
+
+        self.dense_1 = nnx.Linear(in_features=64, out_features=2, rngs=rngs)
+
+    def __call__(self, x: jax.Array):
+        x = self.conv_1(x)
+        x = self.layer_norm_1(x)
+        x = jax.nn.relu(x)
+
+        x = self.conv_2(x)
+        x = self.layer_norm_2(x)
+        x = jax.nn.relu(x)
+
+        x = self.conv_3(x)
+        x = self.layer_norm_3(x)
+        x = jax.nn.relu(x)
+
+        x = jnp.mean(x, axis=(1,))  # global average pooling
+        x = self.dense_1(x)
+        x = jax.nn.softmax(x)
+        return x
+```
+
+```{code-cell} ipython3
+model = MyModel(rngs=nnx.Rngs(0))
+nnx.display(model)
```

Comment:
This cell is producing a GPU warning – let's remove it from the notebook output to avoid distracting readers.

Reply:
Done!

I squashed the PR, it's ready from my side!